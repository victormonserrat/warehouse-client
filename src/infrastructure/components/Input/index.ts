import styled from 'styled-components'

const Input = styled.input`
  box-shadow: 0 0.0625rem 0 ${({ theme }) => theme.colors.grey};
  color: ${({ theme }) => theme.colors.black};
  font-family: ${({ theme }) => theme.fonts.primary};
  font-size: ${({ theme }) => theme.sizes.medium};
  padding: 0.25rem 0;
  width: 100%;

  :focus {
    box-shadow: 0 0.125rem 0 ${({ theme }) => theme.colors.blue};
    color: ${({ theme }) => theme.colors.blue};
  }

  &[aria-invalid='true'] {
    box-shadow: 0 0.125rem 0 ${({ theme }) => theme.colors.red};
    color: ${({ theme }) => theme.colors.red};
  }

  :disabled {
    box-shadow: 0 0.0625rem 0 ${({ theme }) => theme.colors.greyLight};
    color: ${({ theme }) => theme.colors.greyLight};
  }
`

export default Input
