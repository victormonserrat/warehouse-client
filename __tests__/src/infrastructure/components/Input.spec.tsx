import Input from '~/infrastructure/components/Input'
import { render } from '~/tests/utils'

describe('Input', () => {
  it('renders correctly', () => {
    const { container } = render(<Input />)

    expect(container).toMatchSnapshot('Regular')
  })

  it('renders disabled correctly', () => {
    const { container } = render(<Input disabled />)

    expect(container).toMatchSnapshot('Disabled')
  })

  it('renders invalid correctly', () => {
    const { container } = render(<Input aria-invalid />)

    expect(container).toMatchSnapshot('Invalid')
  })
})
