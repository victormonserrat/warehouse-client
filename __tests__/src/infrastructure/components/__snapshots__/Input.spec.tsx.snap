// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly: Regular 1`] = `
<div>
  <input
    class="sc-bczRLJ ejAZYi"
  />
</div>
`;

exports[`Input renders disabled correctly: Disabled 1`] = `
<div>
  <input
    class="sc-bczRLJ ejAZYi"
    disabled=""
  />
</div>
`;

exports[`Input renders invalid correctly: Invalid 1`] = `
<div>
  <input
    aria-invalid="true"
    class="sc-bczRLJ ejAZYi"
  />
</div>
`;
